Toggle platforms to build with grunt cli option

vonbirdie opened the pull request:
Toggle platforms to build with grunt cli option instead of manual toggle in Gruntfile.js. This allows you to invoke grunt like this:

```
$ grunt nodewkbuild
```

And it will build only for the current platform. Or like this:

```
$ grunt nodewkbuild --platforms=win,linux,mac
```

And it will build for the specified platforms in the comma separated list.

This probably shouldn't be merged without fixing the regex for windows. It currently evaluates to true when process.platform is darwin (on os x). I can fix if you want this but just couldn't think of some nice solution for now...

Commit:
Toggle platforms to build with grunt option instead of manual toggle in
Gruntfile.js

Commit:
Remove some obsolete comments

GonzaGonza commented:
Good idea!
Could you do this in the meantime?

```
grunt nodewkbuild --platforms=win,linux32,linux64,mac
```

Those are the platforms. But, if you run the command without the --platforms argument, then it build it for every platform. What do you think?
Thanks!

Commit:
Add support for linux32 and separate win from darwin

Commit:
Document the availability of linux32

vonbirdie commented:
I've added support for linux32 and renamed linux to linux64.

But my reasoning for making it the default to only build for the current platform is that contributors and users probably only want to build for their current platform and making it as easy for them to do so as possible might help reduce the number of incoming questions.

The only ones that i can think of that cares about building it for all platforms would be core developers and maybe some Linux package maintainers that still need to worry about linux32 and both of these groups probably either read the documentation and quickly get accustomed to adding the `--platforms` flag when needed. If you want i could add this

```
grunt nodewkbuild --platforms=all
```

to build for all platforms? Does that sound good?

I can switch it to build for all platforms as default if that's what you want though?

zbrox commented:
Most people will probably build only for their current platform. The --platforms=all option seems reasonable to me.

GonzaGonza commented:
Yes! @luuse and @zbrox i agree, but as i read on @luuse PR he was having problems with a regex detecting the platform.
I will put this PR on hold until someone can resolve the issue detecting each platform.

And again, Thanks!!! nice work!

vonbirdie commented:
@GonzaGonza it should detect all platforms correctly now and everything works as i would expect it to so i'd say it's ready to merge?

Commit:
Add option to build all platforms

vonbirdie commented:
I added the option to build for all platforms with `--platforms=all` but i'd say everything is ready to be merged now :)

GonzaGonza commented:
Nice Work @luuse you rock!

vonbirdie commented:
Thanks :)

zosiu commented:
:+1:
